### PR TITLE
DOC Add intersphinx linking to docs

### LINKED ISSUE #93: DOC Add intersphinx extension
Currently, links to the documentation of other libraries are not resolved, e.g. to scikit-learn. We should fix this by adding the `'sphinx.ext.intersphinx'` extension to `conf.py`, as well as the relevant mappings.

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Fixes #93

With this extension, skops docs can link, for instance, to the sklearn
docs.

On top of adding the extension, I made the following changes:

- Adjust links to external docs where appropriate
- Fix minor issues I found in the docs
- Remove autodocs for type annotations, as they were redundant with the
  docstrings (but less detailed)

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
@skops-dev/maintainers Ready for review. Built docs are [here](https://skops--97.org.readthedocs.build/en/97/).

File: docs/conf.py [hunk 1]
```diff
@@ -80,3 +81,14 @@
...
+
+# See:
+# https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_mapping
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3", None),
+    "numpy": ("https://docs.scipy.org/doc/numpy/", None),
+    "sklearn": ("https://scikit-learn.org/stable/", None),
+    "pandas": ("https://pandas.pydata.org/docs/", None),
+    "joblib": ("https://joblib.readthedocs.io/en/latest/", None),
+    "huggingface_hub": ("https://huggingface.co/docs/huggingface_hub/main/en", None),
```
Comment (adrinjalali) [inline docs/conf.py:93]: does this actually work?

**Review by adrinjalali — APPROVED** [review]
Nice refinements. Merging, more fixes (if any) can be added later. Thanks @BenjaminBossan

File: docs/conf.py (hunk 1 above)
Comment (BenjaminBossan) [reply docs/conf.py:93]: Yes, it works indeed :)

See [here](https://skops.readthedocs.io/en/latest/modules/classes.html#skops.hub_utils.download), at the bottom the link to `huggingface_hub.snapshot_download`

**Pull request merged**